internal/ir/xds.go

```diff
@@ -3143,3 +3144,10 @@ type RequestBuffer struct {
 	// Limit defines the maximum buffer size for requests
 	Limit resource.Quantity `json:"limit" yaml:"limit"`
 }
+
+// ZoneAwareRouting holds the zone aware routing configuration
+// +k8s:deepcopy-gen=true
+type ZoneAwareRouting struct {
+	Enabled bool `json:"enabled" yaml:"enabled"`
```

Comment:
you could do without `Enabled` by leveraging the ptr to decide if its set or not